src/windows/scanner.cc

```diff
@@ -220,15 +220,23 @@ ScanMountpoints(drivelist::com::Connection *const connection,
     if (FAILED(result))
       return InterpretHRESULT(result);
 
-    BOOL isProtected;
-    result = drivelist::volume::GetReadOnlyFlag(path[0], &isProtected);
+    // Turns out a disk can be writable while its volumes can be read-only
+    // and viceversa, so we need to check the writable capabilities of
+    // both the disk and its volumes
+    BOOL writable;
+    result = drivelist::volume::IsDiskWritable(path[0], &writable);
     if (FAILED(result))
       return InterpretHRESULT(result);
+    if (writable && hasFilesystem) {
+      result = drivelist::volume::IsVolumeWritable(path[0], &writable);
+      if (FAILED(result))
+        return InterpretHRESULT(result);
+    }
 
     drivelist::mountpoint_s mountpoint;
     mountpoint.path = path;
     mountpoint.disk = "\\\\.\\PHYSICALDRIVE" + std::to_string(number);
-    mountpoint.readonly = isProtected;
+    mountpoint.readonly = !writable;
     mountpoint.system = path[0] == systemDriveLetter;
     mountpoint.hasFilesystem = hasFilesystem;
```

Comment:
> Does this mean that devices with i.e. both NTFS & HFS+ partitions under Windows with Boot Camp will always be read-only, as the HFS+ driver is read-only on Windows with Boot Camp?

In this case, we will show the `\\.\\PHYSICALDRIVE` entry, which will contain a single volume (e.g. `C:`) that corresponds to the NTFS partition

> So read-only devices with unrecognised file systems are filtered out as well? For example, a read-only SD card with an extfs on it?

Let's say a physical drive is detected as writable. This drive contains a read-only volume without a file-system. The output of drivelist will show a read-only `\\.\PHYSICALDRIVE` without any volumes.

So we filter out volumes without recognized file-systems, but we still take their writable state into consideration.